bouter/embedded/__init__.py

```diff
@@ -28,19 +48,51 @@ def vigor(self, vigor_duration=0.05):
         )
         return self.behavior_log["vigor"]
 
-    def reconstruct_missing_segments(self):
+    def reconstruct_missing_segments(self, continue_curvature=0):
         """ If the tail tip is not tracked throught the whole experiment
         reconstruct the tail sum from the segments that are
 
         """
-        pass
+        utilities.fill_out_segments(
+            self.tail_points_matrix, continue_curvature=continue_curvature
+        )
 
+    @decorators.cache_results
     def bouts(self, vigor_threshold=0.1):
-        bout_path = self.root / self.session_id + "_bouts.h5"
-        if bout_path.is_file():
-            return fl.load(bout_path)
+        # TODO add padding
+        """Extract bouts above threshold.
+        :param vigor_threshold:
+        :return:
+        """
         vigor = self.vigor()
-        bout_starts_ends, cont = utilities.extract_segments_above_threshold(
-            vigor, vigor_threshold
+        return utilities.extract_segments_above_threshold(
```

Comment:
you just want the first element of the returned tuple, continuity is not meaningfull for embedded experiments

Reply:
ok